suite-native/atoms/src/Switch.tsx

```diff
@@ -9,46 +9,54 @@ import Animated, {
 } from 'react-native-reanimated';
 
 import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Color } from '@trezor/theme';
+import { palette } from '@trezor/theme';
 
 import { ACCESSIBILITY_FONTSIZE_MULTIPLIER } from './Text';
 
 type SwitchProps = {
     isChecked: boolean;
     onChange: (value: boolean) => void;
     isDisabled?: boolean; // Functionality of disabled works but styles are not implemented yet (waiting for design)
-    colorChecked?: Color;
-    colorUnchecked?: Color;
 };
 
-const SWITCH_CONTAINER_WIDTH = 44 * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
-const SWITCH_CONTAINER_HEIGHT = 24 * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
+const SWITCH_CONTAINER_BORDER_WIDTH = 1 * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
+const SWITCH_CONTAINER_WIDTH =
+    44 + 2 * SWITCH_CONTAINER_BORDER_WIDTH * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
+const SWITCH_CONTAINER_HEIGHT =
+    24 + 2 * SWITCH_CONTAINER_BORDER_WIDTH * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
+
 const SWITCH_CIRCLE_SIZE = 20 * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
-const SWITCH_CIRCLE_MARGIN = 2 * ACCESSIBILITY_FONTSIZE_MULTIPLIER;
+const SWITCH_CIRCLE_MARGIN = 2 * ACCESSIBILITY_FONTSIZE_MULTIPLIER + SWITCH_CONTAINER_BORDER_WIDTH;
 const SWITCH_CIRCLE_TRACK_WIDTH =
     SWITCH_CONTAINER_WIDTH - SWITCH_CIRCLE_SIZE - SWITCH_CIRCLE_MARGIN * 2;
 
-const switchContainerStyle = prepareNativeStyle(utils => ({
-    width: SWITCH_CONTAINER_WIDTH,
+const switchContainerStyle = prepareNativeStyle<{ isChecked: boolean }>((utils, { isChecked }) => ({
     height: SWITCH_CONTAINER_HEIGHT,
+    width: SWITCH_CONTAINER_WIDTH,
     borderRadius: utils.borders.radii.round,
     flexDirection: 'row',
+    borderWidth: SWITCH_CONTAINER_BORDER_WIDTH,
+    borderColor: 'transparent',
+    extend: [
+        {
+            condition: !isChecked,
+            style: {
+                borderColor: utils.colors.borderElevation0,
+            },
+        },
+    ],
 }));
 
 const switchCircleStyle = prepareNativeStyle(utils => ({
     width: SWITCH_CIRCLE_SIZE,
     height: SWITCH_CIRCLE_SIZE,
-    backgroundColor: utils.colors.backgroundSurfaceElevation1,
+    backgroundColor: palette.darkGray1000,
```

Comment:
why is pallete used instead of color nickname?
```suggestion
    backgroundColor: utils.colors.backgroundNeutralBold,
```

Reply:
This is because the color is the same for both light and dark scheme. `backgroundNeutralBold` makes it dark gray in light mode. 


We dont have any color token like that, because the change is not in the design system, so I directly used palette color which felt better than using `#ffffff` directly.